### PR TITLE
HTML Refactor 

### PR DESCRIPTION
**Pull request by Alek99**
```python
"""Welcome to Reflex! This file outlines the steps to create a basic app."""
from rxconfig import config

import reflex as rx

docs_url = "https://reflex.dev/docs/getting-started/introduction"
filename = f"{config.app_name}/{config.app_name}.py"


class State(rx.State):
    """The app state."""

    pass


def index():
    return rx.fragment(
        rx.el.footer(
            "\n",
            rx.el.div(
                "\n",
                rx.el.div(
                    "\n",
                    rx.el.div(
                        "\n",
                        rx.el.span("Unique", class_name="text-violet-500"),
                        "\n",
                        rx.el.span("Fund", class_name="dark:text-slate-400"),
                        "\n",
                        class_name="w-full text-lg font-extrabold md:w-fit md:text-left",
                    ),
                    "\n",
                    rx.el.p(
                        "\n        UniqueFund provides access to fast and reliable financing solutions for\n        SMEs.\n      ",
                        class_name="text-sm text-slate-600",
                    ),
                    "\n",
                    class_name="flex w-full flex-col space-y-4 text-center md:w-2/5 md:text-left",
                ),
                "\n",
                rx.el.div(
                    "\n",
                    rx.el.div("Company", class_name="text-sm font-semibold"),
                    "\n",
                    rx.el.ul(
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            About\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Terms of Service\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Privacy Policy\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Cookie Policy\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        class_name="text-sm",
                    ),
                    "\n",
                    class_name="w-full text-center text-slate-600 md:w-1/5 md:text-left",
                ),
                "\n",
                rx.el.div(
                    "\n",
                    rx.el.div("Solutions", class_name="text-sm font-semibold"),
                    "\n",
                    rx.el.ul(
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Business Line of Credit\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            SBA Loan\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Revenue Based Financing\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Invoice Factoring\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        class_name="text-sm",
                    ),
                    "\n",
                    class_name="w-full text-center text-slate-600 md:w-1/5 md:text-left",
                ),
                "\n",
                rx.el.div(
                    "\n",
                    rx.el.div("Partnership", class_name="text-sm font-semibold text-slate-600"),
                    "\n",
                    rx.el.ul(
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Loan Partner\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Affiliate\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        rx.el.li(
                            "\n",
                            rx.el.a(
                                "\n            Brand Guideline\n          ",
                                class_name="underline decoration-transparent underline-offset-4 transition hover:decoration-slate-700",
                                href="#",
                                rel="noopener noreferrer",
                                target="_blank",
                            ),
                            "\n",
                            class_name="pt-3",
                        ),
                        "\n",
                        class_name="text-sm",
                    ),
                    "\n",
                    class_name="w-full text-center text-slate-600 md:w-1/5 md:text-left",
                ),
                "\n",
                class_name="container mx-auto flex flex-col items-start space-y-12 px-8 pb-8 pt-12 md:flex-row md:space-y-0 md:space-x-12 md:px-12",
            ),
            "\n",
            rx.el.div(
                "\n    \u00a9\n    \n    2023\n    \n    UniqueFund. All rights reserved.\n  ",
                class_name="container mx-auto border-t border-slate-200 p-8 text-center text-sm text-slate-600 dark:border-slate-900 md:flex-row md:px-12",
            ),
            "\n",
            class_name="mt-28 bg-slate-50 dark:bg-black",
        )
    )

# Add state and page to the app.
app = rx.App()
app.add_page(index)
app.compile()
```

### CODE REVIEW

File: reflex/el/elements/base.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+from reflex.el.element import Element
+from reflex.vars import Var as Var_
+
+
+class BaseHTML(Element):
+    """Base class for common attributes."""
+
+    #  Provides a hint for generating a keyboard shortcut for the current element.
+    access_key: Var_[Union[str, int, bool]]
```
Comment (masenf) [inline reflex/el/elements/base.py:12]: i'd like to read this type as `access_key: Var_[Attribute]` or maybe `Var_[HTMLAttribute]`.

something that reduces the complexity of what my brain has to grok to understand the line.

File: reflex/el/elements/inline.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
+"""Element classes. This is an auto-generated file. Do not edit. See ../generate.py."""
+from typing import Union
+
+from reflex.vars import Var as Var_
+
+from .base import BaseHTML
+
+
+class A(BaseHTML):  # Inherits common attributes from BaseMeta
```
Comment (masenf) [inline reflex/el/elements/inline.py:9]:
```suggestion
class A(BaseHTML):
```

**Review by masenf — DISMISSED** [review]
This is great. I think documenting and maintaining this `rx.el` namespace is going to be good for people who want a truly lightweight app. With no chakra/radix components in use, then they wont be installed at all!

potential conflict with this recent PR: https://github.com/reflex-dev/reflex/pull/2162/commits/e761a049b21878db3e4e0650c912138c4b006480

The `rx.el.element.Element.render` method will still end up accidentally getting `sx` props because they will get added in `_render` and not removed here. I'm thinking that now is probably the time to have `Component` pass `style` as `style` and have `ChakraComponent` pass `style` as `sx`. Then we can stop special casing that here and in the radix tree.

**Comment by Alek99** [comment]
One question should I move the .el into the components directory?

**Comment by masenf** [comment]
> One question should I move the .el into the components directory?

I think it would make more sense in there, but I think we should still expose it as `rx.el` in the API

File: scripts/precompile.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 from reflex.utils import path_ops
 
-from .constants import ELEMENT_TO_PROPS, ELEMENTS
+from ..reflex.components.el.constants import ELEMENT_TO_PROPS, ELEMENTS
```
Comment (picklelo) [inline scripts/precompile.py:17]: We shouldn't need a relative import, does the normal `from reflex.components.el.constants` work here?

File: reflex/components/radix/themes/components.pyi [hunk 4]
```diff
@@ -992,7 +992,7 @@ class TextField(TextFieldRoot, el.Input):
 
         Args:
             *children: The children of the component.
-            size: Text field size "1" - "3"
+            size: Specifies the visible width of a text control
```
Comment (masenf) [inline reflex/components/radix/themes/components.pyi:995]: The inherited HTML element description probably shouldn't take precedent over the description from the class

**Review by masenf — APPROVED** [review]
Looks awesome.

For future work, we still need to get event triggers defined on some of these, but I think we should roll with what is already a big improvement.

--------------

The `Svg` element is essentially a separate XML doc embedded in the HTML, and I don't know how we want to handle that, but as it is, i don't think we will be reverse compiling svg to reflex code anytime soon, nor do i think that's desirable from a perf perspective. We can probably just inline the svg code in the `Svg` element for the reverse compiler in the short term.

**Pull request merged**